File: micronaut-camunda-bpm-feature/transactional-test-data-jdbc/src/test/java/info/novatec/micronaut/camunda/bpm/feature/test/Book.java
package info.novatec.micronaut.camunda.bpm.feature.test;

import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;

/**
 * Sample entity for micronaut-data.
 *
 * @author Lukasz Frankowski
 * @author Tobias Schäfer
 */
@MappedEntity
public class Book {

	@Id
	@GeneratedValue
	private Long id;
	private String title;

	public Book() {
	}

	public Book(String title) {
		this.title = title;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getTitle() {
		return title;
	}
}

File: micronaut-camunda-bpm-feature/transactional-test-data-jdbc/src/test/java/info/novatec/micronaut/camunda/bpm/feature/test/Book.kt
package info.novatec.micronaut.camunda.bpm.feature.test

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity

/**
 * Sample entity for micronaut-data.
 *
 * @author Lukasz Frankowski
 * @author Tobias Schäfer
 */
@MappedEntity
data class Book(val title: String) {
    @Id
    @GeneratedValue
    var id: Long? = null
}